src/actinia_core/processing/actinia_processing/ephemeral/persistent_processing.py

```diff
@@ -324,42 +324,59 @@ def _update_views_in_tgis(self, tgis_db_path):
             con.close()
         del cur
 
-    def _merge_tgis_dbs(self, tgis_db_path_1, tgis_db_path_2):
-        """Merge two tgis sqlite.db files
-
-        Args:
-            tgis_db_path_1(str): path of a tgis sqlite.db file in which the
-                                 other should be merged
-            tgis_db_path_2(str): path of a tgis sqlite.db file which should be
-                                 merged in tgis_db_path_1
-        """
-        con = sqlite3.connect(tgis_db_path_1)
-        con.execute(f"ATTACH '{tgis_db_path_2}' as dba")
-        con.execute("BEGIN")
-
-        table_names1 = [row[1] for row in con.execute(
-            "SELECT * FROM sqlite_master where type='table'")]
-        table_names2 = [row[1] for row in con.execute(
-            "SELECT * FROM dba.sqlite_master where type='table'")]
-
-        # merge databases
-        for table in table_names2:
-            if table == 'tgis_metadata':
-                con.execute(f"DROP TABLE {table}")
-                con.execute(f"CREATE TABLE {table} AS "
-                            f"SELECT * FROM dba.{table}")
-                continue
-            # for example raster_register_xxx tables are not in both dbs
-            if table not in table_names1:
-                con.execute(f"CREATE TABLE {table} AS "
-                            f"SELECT * FROM dba.{table}")
-                continue
-            combine = f"INSERT OR IGNORE INTO {table} SELECT * FROM dba.{table}"
-            con.execute(combine)
-        con.commit()
-        con.execute("detach database dba")
-        if con:
-            con.close()
+    def _change_mapsetname_in_tgistable(
+            self, cur, table_name,
+            source_mapset, target_mapset, skip_columns=[]):
+        columns = [row[0] for row in cur.execute(
+            f"SELECT * FROM {table_name}").description]
+
+        # find PRIMARY KEY
+        selection = [row[1] for row in cur.execute(
+            f"PRAGMA table_info({table_name})") if row[-1] == 1]
+        if len(selection) == 0:
+            primary_key = ""
+        else:
+            primary_key = selection[0]
+
+        for col in columns:
+            if col not in skip_columns:
+                update_statement = f"UPDATE {table_name} SET {col} = " \
+                    f"REPLACE({col}, '{source_mapset}', '{target_mapset}')"
```

Comment:
What if the name of the map and of the source mapset are the same, e.g. `elevation@elevation`? This would then not only replace the mapset name, but also the map name?